WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -245,6 +257,225 @@ class PostCoordinator: NSObject {
         }
     }
 
+    // MARK: - Sync (Drafts)
+
+    /// Returns `true` if post has any revisions that need to be synced.
+    static func isSyncNeeded(for post: AbstractPost) -> Bool {
+        post.original().getLatestRevisionNeedingSync() != nil
+    }
+
+    /// Saves the scratchpad in a local revision scheduled for sync with the server.
+    ///
+    /// - warning: This should only be used for draft posts.
+    ///
+    /// - warning: Work-in-progress (kahu-offline-mode)
+    func setNeedsSync(for revision: AbstractPost) {
+        assert(revision.isRevision(), "Must be used only on revisions")
+        assert(revision.original().status == .draft, "Must be used only with draft posts")
+        revision.isSyncNeeded = true
+        ContextManager.shared.saveContextAndWait(coreDataStack.mainContext)
+
+        let original = revision.original()
+        scheduleSync(for: original)
+    }
+
+    /// Represents an operation for syncing the post and updating it to the
+    /// revision the operation was initialized with.
+    final class SyncOperation {
```

Comment:
FYI I made some large changes to make this part of sync more reliable and readable in the upcoming PRs, so I wouldn't spend too much time on review now.